tiledb/sm/array_schema/array_schema.h

```diff
@@ -255,9 +296,11 @@ class ArraySchema {
    * It assigns values to the members of the object from the input buffer.
    *
    * @param buff The binary representation of the object to read from.
-   * @return Status
+   * @param uri An optional uri object.
+   * @return A new ArraySchema.
    */
-  Status deserialize(ConstBuffer* buff);
+  static shared_ptr<ArraySchema> deserialize(
+      ConstBuffer* buff, const URI& uri = URI());
```

Comment:
I don't believe there's any reason, say polymorphism, to be returning `shared_ptr` here. If a caller wants it, they can call `make_shared` for themselves.

An empty schema (as from a default constructor) makes no sense. Providing an empty URI would need to return a default-constructed schema, so the argument default makes no sense either.